src/main/java/frc/robot/subsystems/ArmSystem.java

```diff
@@ -0,0 +1,58 @@
+package frc.robot.subsystems;
+
+import com.revrobotics.*;
+import edu.wpi.first.wpilibj.DutyCycleEncoder;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.RobotMap;
+
+public class ArmSystem extends SubsystemBase {
+    private final CANSparkMax motor;
+    private final RelativeEncoder neoEncoder;
+    private final DutyCycleEncoder dutyCycleEncoder;
+
+    public ArmSystem() {
+        motor = new CANSparkMax(RobotMap.ARM_MOTOR_PORT, CANSparkLowLevel.MotorType.kBrushless);
+        neoEncoder = motor.getAlternateEncoder(RobotMap.ARM_ENCODER_CPR);
+        dutyCycleEncoder = new DutyCycleEncoder(RobotMap.ARM_ENCODER_PORT);
+    }
+
+    public void moveUp() {
+        if (getDutyCycleEncoderPosition() >= RobotMap.ARM_MAX_ANGLE){
+            stop();
+        } else {
+            motor.set(0.5);
+        }
+    }
+
+    public void moveDown() {
+        if (getDutyCycleEncoderPosition() <= RobotMap.ARM_MIN_ANGLE){
+            stop();
+        } else {
+            motor.set(-0.5);
+        }
+    }
+
+    public void stop() {
+        motor.stopMotor();
+    }
+
+    public double getNeoEncoderVelocity() {
+        return neoEncoder.getVelocity();
+    }
+
+    public double getNeoEncoderPosition() {
+        return neoEncoder.getPosition() / RobotMap.ARM_GEAR_RATIO * 360;
+    }
+
+    public double getDutyCycleEncoderPosition() {
+        return -(dutyCycleEncoder.getAbsolutePosition() - (214 / 360.0))  * 360;
```

Comment:
Add a comment which explains why you are doing this (the unary minus and subtraction). I'm sure it won't be obvious to other people looking at the code.

Also, make `214 / 360.0` into a class constant, something like `ABSOLUTE_ENCODER_ZERO_OFFSET`

Reply:
done